### PR TITLE
docs: add server-routes example

### PR DESCRIPTION
**Pull request by clemcode**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR introduces a `server` section in Examples and adds an example for creating server routes.

### CODE REVIEW

**Comment by clemcode** [comment]
Sorry, forgot to remove the Draft status, should be good!

File: examples/server/routes/app.vue [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<template>
+  <NuxtExampleLayout example="server/routes">
+    <p>Fetched from <pre>/api/mountain</pre>: {{ mountain }}</p>
+  </NuxtExampleLayout>
+</template>
+
+<script setup>
+const { data: mountain } = await useAsyncData('mountain', () => $fetch('/api/mountain').catch(console.trace))
```
Comment (danielroe) [inline examples/server/routes/app.vue:8]: Should we use `useFetch` here or is the aim to show a more complex use-case?

```suggestion
const { data: mountain } = await useFetch('/api/mountain')
```

Comment (clemcode) [reply examples/server/routes/app.vue:8]: We can definitely use `useFetch` indeed, as the focus is more on making a server route.

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
Thanks!

**Pull request merged**